### ISSUE TITLE
[Form templates] - required/optional label placement and screen magnifiers

### ISSUE THREAD

**Issue by AndrewArch** [18F/web-design-standards#1004]
re: https://playbook.cio.gov/designstandards/form-templates/

Your label positioning - placing required/optional at the RHS of the text-entry box, cf. the description of what they are to enter positioned at the LHS - means that some people won't notice "required/optional" as it's not associated with (adjacent to) the label. More particularly many screen-magnifiers users will likely miss it completely until they get an error message depending on their degree of magnification as they scroll down the LHS of a page. 

However, I'd be curious if your usability testing testing showed otherwise and would like to see the findings if you can share them.

**Comment by Drakenhart** [comment]
I also would like to know what the usability testing was on this, as we are using a simple red asterisk currently.

Forms, if coded as required, will voice "required" for JAWS users, so our testing has shown. We will be doing further testing of our forms at a later date, but any input now on this matter would be very beneficial.

**Comment by carodew** [comment]
Previous user research has shown that spelling out the words "required" or "optional" are easier for users to understand than just the asterisk, which is true for sighted users regardless of whether the form is coded correctly for screen readers. Sometimes asterisks mean a field is optional and sometimes it's required. This is based on recommendations from Luke Wroblewski's book _Web Form Design: Filling in the Blanks_. 

I also (personally) think that coloring the asterisk (or label) red is a little too close visually to an error (which are usually also red) and could impede scanning through the form to look for error validation. 

We recommend labeling the fields as required or optional based on which is fewer, although hopefully if you've designed your form right, there are always going to be fewer optional fields (because you aren't asking for information you don't need). Also, there's been other research that shows people are more likely to complete a form when the optional fields are labeled rather than the required. 

I think it's worth considering the placement of the label in re to screen magnifiers. We specifically did not position that label to the far right of the form fields for that reason, but I'm open to exploring if the required/optional labels should immediately follow the name label (for example, First Name), but the trade off could be it makes it harder for sighted users to scan the form and quickly see what is required or not. That might be an ok trade off. 

We will be working on building our our form templates at some point in the next phase and will definitely consider this feedback. None of the advice we (or any other pattern library, for that matter) should trump your own usability testing with your own users.

**Comment by Drakenhart** [comment]
People are more likely to complete a form when the optional fields are labeled rather then the required? Do you have a link to this research. I don't think I've come across it yet to date. Also thank you for the book reference as well. It is appreciated.

In regards to the other point of quick scan-ability of the form. Perhaps a typographic treatment of the Required and Optional labels might be in order if they are moved to the LHS so they are more easily noticed?

**Comment by carodew** [comment]
Check out this article: http://uxmovement.com/forms/why-users-fill-out-less-if-you-mark-required-fields/

I may have slightly misspoke – it seems that people are more likely to fill out all the fields in a form if you mark the optional fields. But ultimately it's whatever has fewer fields – required or optional. Although hopefully you've eliminated as many optional fields as possible.

**Comment by Drakenhart** [comment]
@carodew Indeed. Thanks again.

**Comment by AndrewArch** [comment]
Here's some GDS discussion on the issue - https://userresearch.blog.gov.uk/2014/07/30/reversing-a-circle-of-mistrust-2/
